### PR TITLE
Viger los gos extension

### PR DESCRIPTION
**Pull request by Vigerus**

### CODE REVIEW

**Review by mostlikely4r — COMMENTED** [review]
Mostly nitpick stuff to prevent some (unneeded) code/functions. Functionally it looks ok.

File: playerbot/ChatHelper.cpp [hunk 1]
```diff
@@ -969,3 +969,28 @@ void ChatHelper::replaceSubstring(std::string& str, const std::string& oldStr, c
         pos = str.find(oldStr, pos + newStr.length());
     }
 }
+
+bool ChatHelper::startswith(const std::string& str, const std::string& prefix)
```
[the remark this answers is not in the document]
Comment (Vigerus) [reply playerbot/ChatHelper.cpp:973]: Definetly. I just came with assumption that we have ChatHelper with all those methods for better code reading. the compiler would optimize it anyway.

File: playerbot/ChatHelper.cpp [hunk 2]
```diff
@@ -969,3 +969,28 @@ void ChatHelper::replaceSubstring(std::string& str, const std::string& oldStr, c
...
+
+bool ChatHelper::startswith(const std::string& str, const std::string& prefix)
+{
+   if (prefix.length() > str.length()) {
+      return false;
+   }
+   return str.compare(0, prefix.length(), prefix) == 0;
+}
+
+std::string ChatHelper::trim(const std::string& str)
```
[the remark this answers is not in the document]
Comment (Vigerus) [reply playerbot/ChatHelper.cpp:981]: nope, i can remove it. same reasoning as above, as afaik no c++ standard for trim yet, only boost.

File: playerbot/strategy/actions/GoAction.cpp [hunk 3]
```diff
@@ -282,15 +300,19 @@ bool GoAction::TravelTo(TravelDestination* dest, Player* requester) const
     }
 }
 
-bool GoAction::MoveToGo(std::string& param, Player* requester)
+//todo: could be a part of bot API
+inline WorldLocation GameObjectToBotLocation(const GameObject& go)
```
[the remark this answers is not in the document]
Comment (Vigerus) [reply playerbot/strategy/actions/GoAction.cpp:304]: The reason i included the 0.5f shift in the method was that I understood 0.5f as a standard AI OFFSET. I even wanted to make a define for it or constexpr. Doing +0.5f around without any info is a bit confusing.

File: playerbot/strategy/actions/GoAction.cpp [hunk 4]
```diff
@@ -301,7 +323,13 @@ bool GoAction::MoveToGo(std::string& param, Player* requester)
 
             std::ostringstream out; out << "Moving to " << ChatHelper::formatGameobject(go);
             ai->TellPlayerNoFacing(requester, out.str());
-            return MoveNear(bot->GetMapId(), go->GetPositionX(), go->GetPositionY(), go->GetPositionZ() + 0.5f, ai->GetRange("follow"));
+            
+            WorldLocation location = GetNearLocation(GameObjectToBotLocation(*go), INTERACTION_DISTANCE);
```
[the remark this answers is not in the document]
Comment (Vigerus) [reply playerbot/strategy/actions/GoAction.cpp:327]: I was actually pretty confused seeing that the interface accepts (x,y,z) and then internally translates it back to WorldLocation / WorldPosition. 

We could ofc stick to one form - WorldPosition can be a preffered way.

File: playerbot/strategy/actions/MovementActions.cpp [hunk 5]
```diff
@@ -45,6 +45,15 @@ bool MovementAction::isUseful()
     return !ai->HasStrategy("stay", ai->GetState());
 }
 
+WorldLocation MovementAction::GetNearLocation(const WorldLocation& location, float distance)
```
[the remark this answers is not in the document]
Comment (Vigerus) [reply playerbot/strategy/actions/MovementActions.cpp:48]: definetly - i copied the code mostly from MoveNear

File: playerbot/strategy/actions/MovementActions.cpp [hunk 6]
```diff
@@ -267,6 +276,11 @@ bool MovementAction::FlyDirect(WorldPosition &startPosition, WorldPosition &endP
 #endif
 }
 
+bool MovementAction::MoveTo(const WorldLocation& location, bool idle, bool react, bool noPath, bool ignoreEnemyTargets)
```
[the remark this answers is not in the document]
Comment (Vigerus) [reply playerbot/strategy/actions/MovementActions.cpp:279]: sure. The reason again was simple - i would replace the x,y,z one with this one, but yeah.

**Review by mostlikely4r — APPROVED** [review]

**Pull request merged**